src/main/java/parser/Parser.java

```diff
@@ -42,9 +42,40 @@ public static Command parse(String userInput) throws DukeException {
                 }
                 if (taskFeatures.isEmpty()) {
                     throw new DukeException(DukeException.EMPTY_USER_DESCRIPTION());
-                } else {
-                    return new AddCommand(command, taskFeatures, null, null, null);
+                } 
+                else {
+                    checkType = "/between";
+                    String dateTimeFromUser;
+                    String taskDescription = taskFeatures.split(checkType, 2)[0].trim();
+                    try {
+                        dateTimeFromUser = taskFeatures.split(checkType, 2)[1].trim();
+                    }catch (ArrayIndexOutOfBoundsException e)
```

Comment:
Program flow should never be controlled by exceptions handling, it's bad practice and expensive.
For now, we can leave it in like this but we need to eventually change this code into something better

Reply:
Correction: This needs to be fixed now